harness/determined/launch/deepspeed.py

```diff
@@ -19,7 +20,34 @@
 from determined.common import api
 from determined.common.api import certs
 
-hostfile_path = "/tmp/hostfile.txt"
+# When the task container uses "/tmp" from the host, having a file with a
+# well-known name in a world writable directory is not only a security issue,
+# but it can also cause a user's experiment to fail due to the file being owned
+# by another user.  Create the file securely with a random name to avoid
+# file name clashes between two different experiments.
+temp_hostfile = tempfile.NamedTemporaryFile(prefix="/tmp/hostfile-", suffix=".txt", delete=False)
```

Comment:
probably should just move this logic into main and pass it around instead. i'm not really a python expert, but having stuff that can fail that gets evaluated at import time seems weird if it isn't 100% necessary.